### PR TITLE
feat: update object change class

### PR DESCRIPTION
**Pull request by ilya-zhuravel**

### CODE REVIEW

File: src/model/history/update-object-change.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      if (overlap > 0) {
+        endA += overlap;
+        endB += overlap;
+      }
+      transaction.replace(start, endB, nextState.doc.slice(start, endA));
+    }
+    return transaction;
+  }
+
+  applyChange(manuscript: Manuscript): Manuscript {
```
Comment (Cooryd) [inline src/model/history/update-object-change.ts:72]: I'd just return the `Manuscript` from this until we can write a full set of unit test for the implementation. Or you can add unit tests for this as part of this PR, up to you (I'd do separately to make code change smaller if I were you though)

File: src/model/history/update-object-change.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    const updatedSection = this.differences.reduce((acc: T, diff) => {
+      const newAcc = cloneManuscript(acc) as T;
+      deepDiff.applyChange(newAcc, acc, diff);
+      return newAcc;
+    }, originalSection);
+
+    return set(cloneManuscript(manuscript), this.path, updatedSection);
+  }
+
+  rollbackChange(manuscript: Manuscript): Manuscript {
```
Comment (Cooryd) [inline src/model/history/update-object-change.ts:83]: Do we need rollback on server? I've left them implementation of this for everything else as just a `console.log` then returning the input manuscript. If this is left in though we need more unit tests for it.

File: test/unit/model/history/update-object-change.test.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  describe('rollbackChange', () => {
+    // TODO: update this when / if implimented
+    it('returns Manuscript', () => {
+      const updateObjChange = UpdateObjectChange.createFromTwoObjects('somepath', {}, {prop: 'value'});
+      const manuscript = {} as unknown as Manuscript;
+      expect(updateObjChange.rollbackChange(manuscript)).toEqual(manuscript);
+    })
+  });
+  describe('isEmpty', () => {
+    // TODO: update this when / if implimented
```
Comment (Cooryd) [inline test/unit/model/history/update-object-change.test.ts:123]: can remove this comment now it's implemented.

File: src/model/history/update-object-change.ts (hunk 2 above)
Comment (ilya-zhuravel) [reply src/model/history/update-object-change.ts:83]: I don't think we need these actually. Can't think of a use case for these on the server.  We can always add them later

**Review by Cooryd — APPROVED** [review]

**Pull request merged**